packages/inferno-redux/src/components/Provider.ts

```diff
@@ -1,35 +1,42 @@
+import {isNullOrUndef, toArray} from 'inferno-shared';
+
 import Component from 'inferno-component';
-import {
-	isNullOrUndef,
-	toArray
-} from 'inferno-shared';
-import { warning } from './utils';
+import { Store } from 'redux';
+import { warning } from '../utils/warning';
 
 let didWarnAboutReceivingStore = false;
-function warnAboutReceivingStore() {
+const warnAboutReceivingStore = () => {
 	if (didWarnAboutReceivingStore) {
 		return;
 	}
+
 	didWarnAboutReceivingStore = true;
 
 	warning(
 		'<Provider> does not support changing `store` on the fly.'
 	);
-}
+};
+
+export interface Props {
+	store: Store<any>;
+	children: any;
+};
 
-export default class Provider extends Component<any, any> {
-	store: any;
+export class Provider extends Component<Props, any> {
+	static displayName = 'Provider';
+	private store: Store<any>;
 
-	constructor(props, context?: any) {
+	constructor(props: Props, context: any) {
 		super(props, context);
 		this.store = props.store;
 	}
 
 	getChildContext() {
-		return { store: this.store };
+		return { store: this.store, storeSubscription: null };
 	}
 
 	render() {
+		// TODO: Maybe not allocate an array here for no reason?
 		if (isNullOrUndef(this.props.children) || toArray(this.props.children).length !== 1) {
```

Comment:
You don't want length check to be only on dev. Basically this implements React.Children.only